pages/_app/index.jsx

```diff
@@ -178,18 +200,45 @@ class App extends NextApp {
     return null
   }
 
+  Login = () => {
+    const { router } = this.props
+    const param = 'logout' in router.query ? 'logout' : 'loading'
+
+    return (
+      <iframe
+        ref={this.loginIframeRef}
+        title="Login Form"
+        src={`/login.html?${param}`}
+        className={styles.loginIframe}
+        onLoad={() => {
+          const { isAuthorized, loginInProcess } = this.state
+          if (!isAuthorized && !loginInProcess) {
+            this.loginIframeRef.current.contentWindow.postMessage(
+              'login-finished',
+              window.location.origin
+            )
+          }
+        }}
+      />
+    )
+  }
+
   render() {
     const { store } = this
     const { isAuthorized } = this.state
     const { Component, pageProps } = this.props
+    const { Login } = this
 
     if (store == null || !isAuthorized) {
       return (
-        <iframe
-          title="Login Form"
-          src="/login.html?loading"
-          className={styles.loginIframe}
-        />
+        <>
+          <Application
+            dataProvider={undefined}
+            activity="overview"
+            onClick={this.handleNavigation}
+          />
+          <Login />
```

Comment:
Why you did not pass this as an Application's child?

Reply:
I think I answered it here https://github.com/oacore/dashboard/pull/186#discussion_r392923262

Reply:
I still do not understand why this is not a child.

Reply:
What do you mean? Like login should be a new component?

Reply:
The _Login_ is a ~new~ component. You render it along with the _Application_ as a sibling. I am just curious why do not you render it as an Application's child?

Reply:
Oh yes. The idea behind it is that I wanted to have blurred submenu too. (at least for logout). Also now the log in form is centered both vertically and horizontally. It could be achieved as a child too I guess.